VultisigApp/VultisigApp/View Models/SendCryptoVerifyViewModel.swift

```diff
@@ -43,6 +43,52 @@ class SendCryptoVerifyViewModel: ObservableObject {
         )
     }
     
+    func blowfishSolanaTransactionScan(tx: SendTransaction, vault: Vault) async -> BlowfishResponse? {
+        
+        var keysignPayload: KeysignPayload?
+        
+        do {
+            let chainSpecific = try await blockChainService.fetchSpecific(for: tx, sendMaxAmount: tx.sendMaxAmount, isDeposit: tx.isDeposit, transactionType: tx.transactionType)
+            let keysignPayloadFactory = KeysignPayloadFactory()
+            keysignPayload = try await keysignPayloadFactory.buildTransfer(coin: tx.coin,
+                                                                           toAddress: tx.toAddress,
+                                                                           amount: amount(for:tx.coin,tx:tx),
+                                                                           memo: tx.memo,
+                                                                           chainSpecific: chainSpecific,
+                                                                           vault: vault
+            )
+            
+            
+            var zeroSignedTransaction: String = .empty
+            if let payload = keysignPayload {
+                zeroSignedTransaction = try SolanaHelper.getZeroSignedTransaction(vaultHexPubKey: vault.pubKeyEdDSA, vaultHexChainCode: vault.hexChainCode, keysignPayload: payload)
+                
+            }
+            
+            
+            return try await BlowfishService.shared.blowfishSolanaTransactionScan(
+                fromAddress: tx.fromAddress, zeroSignedTransaction: zeroSignedTransaction
+            )
+            
+        } catch {
+            switch error {
+            case KeysignPayloadFactory.Errors.notEnoughBalanceError:
```

Comment:
KeysignPayloadFactory.Errors implement LocalizedError. You can use: self.errorMessage = error.localizedDescription

Reply:
Done